pkg/suggestion/v1alpha3/hyperopt/base_hyperopt_service.py

```diff
@@ -0,0 +1,150 @@
+import hyperopt
+import numpy as np
+import logging
+
+from pkg.suggestion.v1alpha3.internal.search_space import *
+from pkg.suggestion.v1alpha3.internal.trial import *
+
+logger = logging.getLogger("BaseHyperoptService")
+
+class BaseHyperoptService(object):
+    def __init__(self, algorithm_name="tpe"):
+        if algorithm_name == 'hyperopt-tpe':
+            self.hyperopt_algorithm = hyperopt.tpe.suggest
+        elif algorithm_name == 'hyperopt-random':
+            self.hyperopt_algorithm = hyperopt.rand.suggest
+        # elif algorithm_name == 'hyperopt-anneal':
+        #     self.hyperopt_algorithm = hyperopt.anneal.suggest_batch
+        # elif algorithm_name == 'hyperopt-mix':
+        #     self.hyperopt_algorithm = hyperopt.mix.suggest
+        else:
+            logger.error("Failed to create the algortihm: %s", algorithm_name)
+            self.hyperopt_algorithm = None
```

Comment:
I think we should raise exception here. since in getSuggestions we always take self.hyperopt_algorithm as valid one

Reply:
SGTM